### PR TITLE
T5

### PR DESCRIPTION
**Pull request by cockochan**
Thank you for contributing! To help the review process, please provide the following:

### Proposal
Theese are the changes from t4 merged with master with solved conflicts. Something happened to fetch links should work well if links provide data

### Related
Provide links to any associated pull requests or issues. You can use `#123` to link to a PR or issue in this repository, or `user/repo#123` to link to other repositories.

### Checklist

- [x] I have read the [code of conduct] and [contributing guide]
- [x] I have made this pull request to the `master` branch
- [x] I have run all of the automated validation using `npm run ship`
- [x] I have added myself to the `"contributors"` list in the `package.json` (or do not want to)

[code of conduct]: https://github.com/textbook/starter-kit/blob/master/.github/CODE_OF_CONDUCT.md
[contributing guide]: https://github.com/textbook/starter-kit/blob/master/.github/CONTRIBUTING.md

### CODE REVIEW

**Review by yusuf963 — COMMENTED** [review]
good job

File: client/src/Components/Mentors.js [hunk 1]
```diff
@@ -5,81 +5,113 @@ import {
...
 	const addQuestion =(e)=>{
 		setNewQuizz({ ...newQuizz,
-			questions:[...newQuizz.questions,e.target.value],
+			questions_id:[...newQuizz.questions_id,e.target.value],
 		},
 		);
+	};
+	const submitNewQ =()=>{
+		if(newQuizz.name.length<8){
+
```
Comment (naderakhgari) [inline client/src/Components/Mentors.js:50]: newQuiz is better than newQuizz

File: client/src/Components/Mentors.js [hunk 2]
```diff
@@ -5,81 +5,113 @@ import {
...
 		return(<div className='row'>
 			<div className='col-9 cardBlock'>
-				{fetchedQuestions.map((quest)=>
+				{props.questions.map((quest)=>
 					<div className='col-3 card'>
-						<input className="quizCardCheckbox" type="checkbox" id="horns" name="horns" value={quest.id} onChange={addQuestion} />
-						<label htmlFor="horns">add to quiz</label>
+
+						<button id={quest._id} value={quest._id} onClick={addQuestion}>add to quiz</button>
+
```
Comment (naderakhgari) [inline client/src/Components/Mentors.js:94]: quest doesn't have meaning maybe question

File: client/src/Components/Results.js [hunk 3]
```diff
@@ -0,0 +1,23 @@
...
+
+export default function Results(props) {
+
+	if(props.quizes){
+		return (
+			<div className='col-12'>
+				<h1>Welcome to QuizzTime</h1>
+
+				<select name="quizzez" id="quizzez" onChange={(e) => quizzChosen(e.target.value)}>
+					<option >select a quiz</option>
```
Comment (naderakhgari) [inline client/src/Components/Results.js:11]: quizzez is not right, you can change it to quizzes

File: client/src/Components/Students.js [hunk 4]
```diff
@@ -2,44 +2,44 @@ import React, { useState, useEffect } from "react";
...
+      {fetchedData.questions_id ? (
+        <Questions
+          fetchedData={fetchedData}
+          quizId={quizId}
+          quizName={quizName}
+        />
+      ) : null}
+    </div>
+  );
 };
```
Comment (naderakhgari) [inline client/src/Components/Students.js:44]: Constantin please don't change my code. PLEASE PLEASE PLEASE
I mean when you make a pull request just add the files which is related to your task

File: client/webpack/common.config.js [hunk 5]
```diff
@@ -19,6 +19,9 @@ module.exports = {
 			},
 		],
 	},
+	output: {
+		publicPath: "/",
+	},
```
Comment (naderakhgari) [inline client/webpack/common.config.js:24]: why did you change this file ?

File: client/webpack/prod.config.js [hunk 6]
```diff
@@ -25,6 +25,7 @@ module.exports = merge(common, {
 	},
 	plugins: [
 		new HtmlWebpackTagsPlugin({
+			usePublicPath: false,
```
Comment (naderakhgari) [inline client/webpack/prod.config.js:28]: why did you change this file ?

File: package.json [hunk 7]
```diff
@@ -67,6 +67,7 @@
     "cors": "^2.8.5",
     "dayjs": "^1.8.34",
     "dotenv": "^8.2.0",
+    "env": "0.0.2",
```
Comment (naderakhgari) [inline package.json:70]: you do not need this env inside package.json.
you shouldn't add all changes to master just add your file !

**Review by naderakhgari — CHANGES_REQUESTED** [review]

**Review by naderakhgari — CHANGES_REQUESTED** [review]
please just add your file while make a pull request if you don't know what I mean please let me know!

File: client/src/Components/Mentors.js [hunk 8]
```diff
@@ -5,81 +5,113 @@ import {
...
+				let found = (props.questions.find((question)=>question._id==selId));
 				newQuizQuestions.push(found);
-
-
 				setNewQuizzQuestions(newQuizQuestions);
-
 			}
 			);
-		}
-	},[newQuizz]);
```
Comment (yusuf963) [inline client/src/Components/Mentors.js:38]: better to assign  a better variableName

File: client/webpack/common.config.js (hunk 5 above)
Comment (cockochan) [reply client/webpack/common.config.js:24]: that's John Sharpe's bufix foe some heroku stuff

File: client/webpack/prod.config.js (hunk 6 above)
Comment (cockochan) [reply client/webpack/prod.config.js:28]: Same John's fix

File: client/src/Components/Students.js (hunk 4 above)
Comment (cockochan) [reply client/src/Components/Students.js:44]: Yes. You are right this students component i should not even push that was a placeholder

File: client/src/Components/Results.js (hunk 3 above)
Comment (cockochan) [reply client/src/Components/Results.js:11]: I agree. I sometimes have weird variable names because i try not to have same variable names even between components

File: client/src/Components/Mentors.js (hunk 2 above)
Comment (cockochan) [reply client/src/Components/Mentors.js:94]: same here. that's to avoid repeating naming

File: client/src/Components/Mentors.js (hunk 1 above)
Comment (cockochan) [reply client/src/Components/Mentors.js:50]: agree. i suggest changing this on running code when we connect the parts

File: client/src/App.js [hunk 9]
```diff
@@ -1,64 +1,84 @@
...
-						<Mentors />
+						<Mentors questions={questions} quizes={quizes} />
+					</Route>
+					<Route exact path="/Results">
+						<Results questions={questions} quizes={quizes} />
 					</Route>
 					<Route exact path="/Students">
-						{questionData.length >0?<Students questionData={questionData} />:<p>Loading...</p>}
+						<Students questions={questions} quizes={quizes} />
 					</Route>
```
[the remark this answers is not in the document]
Comment (cockochan) [reply client/src/App.js:73]: Agree. Just one suggestion. If you pass later  questionData={questions} this should make same thing in your component as props.questionData as before and our code should run

File: client/src/App.js [hunk 10]
```diff
@@ -1,64 +1,84 @@
...
-import Questions from "../src/mockData/Questions.json";
 export function App() {
-	const [questionData, setQuestionData]=useState({});
-	const [route, setRoute]=useState("quizzes");
-	useEffect(()=>{
-		fetch(`http://localhost:3100/api/${route}`)
+	const [questions, setQuestions] = useState(null);
+	const [quizes, setQuizes] = useState(null);
+	if (questions !== null && quizes !== null) {
+	}
```
[the remark this answers is not in the document]
Comment (cockochan) [reply client/src/App.js:14]: that's basically preveniting from crashing- a primitive props validation. if there are no props it will prevent a part of code from execution as if it starts with no props - app will crash becaye before fetches and State is set props will be null

File: client/src/App.js [hunk 11]
```diff
@@ -1,64 +1,84 @@
...
+import Results from "./Components/Results.js";
 
-import Questions from "../src/mockData/Questions.json";
 export function App() {
-	const [questionData, setQuestionData]=useState({});
-	const [route, setRoute]=useState("quizzes");
-	useEffect(()=>{
-		fetch(`http://localhost:3100/api/${route}`)
+	const [questions, setQuestions] = useState(null);
+	const [quizes, setQuizes] = useState(null);
```
[the remark this answers is not in the document]
Comment (cockochan) [reply client/src/App.js:12]: i think i have quizzez  in some other components propose to change on live code just to make sure it will  not break

**Review by yusuf963 — APPROVED** [review]

**Review by naderakhgari — APPROVED** [review]

**Review by naderakhgari — APPROVED** [review]

File: package.json (hunk 7 above)
Comment (cockochan) [reply package.json:70]: i guess it's from when we did npm install env together on a call by mistake when it should be dotenv package
